Source/WebCore/testing/MockGamepadProvider.h

```diff
@@ -59,6 +61,9 @@ class MockGamepadProvider : public GamepadProvider {
     void gamepadInputActivity();
 
     Vector<PlatformGamepad*> m_connectedGamepadVector;
+    // FIXME: Use HashMap<WeakPtr<GamepadProviderClient>, HashSet<WeakPtr<PlatformGamepad>>>
```

Comment:
Adding new code with raw pointers in containers like this is really scary, even with a FIXME. This goes against the direction we want to take our code towards.

Reply:
I agree, but there are other places in the production code which do the same thing today because these classes don't support WeakPtr.

I think we should have a separate change so that `GamepadProviderClient` and `PlatformGamepad` support WeakPtr, and change all those places?

Reply:
Ok.

Reply:
Previously discussed here with @rniwa:  <https://github.com/WebKit/WebKit/pull/8648#discussion_r1072974143>